Revit_Core_Engine/Modify/Insulation.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.DB.Mechanical;
+using Autodesk.Revit.DB.Plumbing;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Add or update existing insulation of the duct or pipe with given inuslation type and thickness.")]
+        [Input("host", "Host element to add or update the insulation on.")]
+        [Input("insulationType", "Type of insulation to be added or updated on the host object. Null value removes host insulation if exist.")]
+        [Input("insulationThickness", "Thickness of insulation to be added or updated on the host object.")]
+        [Output("insulation", "Insulation that has been added or updated on the host element.")]
+        public static InsulationLiningBase Insulation(this Element host, ElementType insulationType, double insulationThickness)
```

Comment:
There are 2 issues with this method signature I'm afraid:
- `Modify` methods should return `void` or the modified object, so I think the most sensible approach here would be to return `void` (assuming that you are not using the newly created insulation object further downstream) - this will rule out my other comment on this PR
- I would rename the method to `SetInsulation` to follow the general pattern (please note this may affect other PRs)